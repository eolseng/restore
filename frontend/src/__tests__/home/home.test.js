import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../../components/pages/home/home';
import {BrowserRouter} from "react-router-dom";

afterEach(cleanup)

test('should take snapshot', () => {
    const {asFragment} = render(
        <BrowserRouter>
            <Home/>
        </BrowserRouter>);

    expect(asFragment(<Home/>)).toMatchSnapshot()
});

test('display restore', () => {

    const { getByText } = render(<BrowserRouter><Home /></BrowserRouter>);
    const h1 = getByText(/Restore - pant dine kl√¶r i dag/);
    expect(h1).toBeInTheDocument();
});


