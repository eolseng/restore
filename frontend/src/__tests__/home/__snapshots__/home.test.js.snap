// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-wrapper"
  >
    <div
      class="above-fold container-fluid"
    >
      <a
        class="above-fold-link"
        href="/login"
      >
        <span>
          Logg inn / Registrer deg
        </span>
      </a>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="above-fold-text-wrapper col-sm-6"
          >
            <h1>
              Restore - pant dine klær i dag
            </h1>
            <p>
              Send inn eller lever dine brukte klær og motta poeng i din profil. Bruk poengene på nye eller brukte klær hos våre samarbeidspartnere. Spar tiden du ellers ville brukt på å undersøke priser og selge dine brukte klær til andre. 
              <br />
            </p>
            <b>
              Vi gjør jobben for deg- du sender inn brukte klær og mottar poeng til å kjøpe nye klær!
            </b>
            <div
              class="button-container"
            >
              <a
                class="button-link"
                href="/restore"
              >
                <button
                  class="cta-button"
                >
                  Start nå
                </button>
              </a>
            </div>
          </div>
          <div
            class="above-fold-img-wrapper col-sm-6"
          >
            <img
              alt="Hero"
              src="[object Object]"
            />
          </div>
        </div>
      </div>
      <a
        class="down-icon-container"
        href="#anchor"
      />
    </div>
    <div
      class="under-fold"
    >
      <div
        class="container-fluid header-wrapper"
        id="anchor"
      >
        <div
          class="container header-container"
        >
          <div
            class="header-left"
          >
            <span
              class="active-nav-button"
            >
              Slik fungerer det
            </span>
            <span>
              Våre tjenester
            </span>
            <span>
              Vår visjon
            </span>
          </div>
          <div
            class="header-right"
          >
            <a
              class="header-link"
              href="/restore"
            >
              <button
                class="cta-button"
              >
                Start nå
              </button>
            </a>
            <a
              class="header-link header-link-user"
              href="/login"
            >
              <span>
                Logg inn
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="container-fluid content-wrapper"
      >
        <div
          class="container content-container"
        >
          <div
            class="row content"
          >
            <div
              class="img-container col-sm-6"
              id="bluprint-img"
            >
              <img
                alt="How it works"
                src="[object Object]"
              />
            </div>
            <div
              class="text-wrapper col-sm-6"
              id="bluprint-text"
            >
              <h3>
                Slik fungerer det
              </h3>
              <ol>
                <li>
                  Opprett anonym profil
                </li>
                <li>
                  Få verdivurdering på klær på forhånd
                </li>
                <li>
                  Send inn/lever klær
                </li>
                <li>
                  Motta poeng i Store Credit i din profil
                </li>
              </ol>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
